### DISCUSSION TITLE
Angular/Nest cli commands for Nuxi

### DISCUSSION THREAD

**Discussion by Baroshem** [nuxt/nuxt#17802] [Ideas]
Hey guys!

I am using Nest.js on a daily basis and really like the CLI commands that are shipped with the Nest CLI (That are not only used for creating new project but also for creating new elements in your project like controllers or modules).

https://docs.nestjs.com/cli/usages

What do you think about having such a feature as an extension for Nuxi? It could work like this:

`nuxi generate endpoint test`

which will result in creating a new file inside /server/api folder called test with a basic file structure like following:

`export default (req, res) => 'Test works!'`

It could also work for components (vue 3 with script setup), pages, layouts, middlewares etc.

So the basic command structure would be like this:

`nuxi generate (or 'g' for shorthand) <SCHEMATIC> <NAME> <?OPTIONS>`

Let me know what do you think about it.

I can prepare a Proof of Concept if you decide that it's worth to give it a try :)

**Comment by Baroshem** [comment]
nuxt/nuxt#17785

**Reply by Baroshem** [reply]
@danielroe @pi0 any thoughts on that?

**Reply by pi0** [reply]
Hi @Baroshem. Sure actually it is a really nice idea and we would love to add this feature set. This might need to be aligned with nuxt devtools btw which is mainly why we didn't add to plan yet.

**Reply by Baroshem** [reply]
I see, should I create this Proof of Concept PR to nuxi or should I wait until the release of nuxt devtools?

**Reply by Baroshem** [reply]
Bump

**Comment by pi0** [comment]
Thank you so much @Baroshem for the initiative and for making it happen! Let's track progress in #3902.
